### PR TITLE
Fixed #28581 -- Pagination topic docs restructuring

### PR DESCRIPTION
**Pull request by luto**
https://code.djangoproject.com/ticket/28581

### CODE REVIEW

File: docs/index.txt [hunk 1]
```diff
@@ -306,7 +306,7 @@ applications:
 * :doc:`Logging <topics/logging>`
 * :doc:`Sending emails <topics/email>`
 * :doc:`Syndication feeds (RSS/Atom) <ref/contrib/syndication>`
-* :doc:`Pagination <topics/pagination>`
+* :doc:`Pagination <topics/pagination>` as well as the :doc:`Paginator <ref/paginator>` class
```
Comment (claudep) [inline docs/index.txt]: I'm not sure if we need to link to the ref page here. If the topics page has the link at appropriate place(s), isn't that sufficient?

Comment (luto) [reply docs/index.txt]: yep, you are right. I removed the additional link. Thanks!

**Comment by rixx** [comment]
Looks good to me. 👍

File: docs/ref/index.txt [hunk 2]
```diff
@@ -31,3 +31,4 @@ API Reference
    utils
    validators
    views
+   paginator
```
Comment (timmartin) [inline docs/ref/index.txt]: It looks like this list is supposed to be in alphabetical order ("Built-in Views" violates this, but it looks like a mistake). I suggest you move this to the correct place.

File: docs/topics/pagination.txt [hunk 3]
```diff
@@ -2,15 +2,68 @@
...
+
+    from django.views.generic import ListView
+    from books.models import Publisher
+
+    class PublisherList(ListView):
+        paginate_by = 2
+        model = Publisher
+
+Assuming you were able to view the full list of publishers before this change,
+the example above already works! The only thing your users will be missing is
```
Comment (timmartin) [inline docs/topics/pagination.txt]: I'm not sure about this sentence. There's no explicit example of what it looked like before the change, and surely the code as specified will work whether or not it was correct before it was modified. I'd probably just say something simpler like "This is the only change necessary to the view code." Personally I prefer it without an exclamation mark, though I don't think it's actually wrong.

File: docs/topics/pagination.txt [hunk 4]
```diff
@@ -2,15 +2,68 @@
...
+a way to get to the next or previous page comfortably.
```
Comment (timmartin) [inline docs/topics/pagination.txt]: "comfortably" seems like an odd word. I'd be tempted to leave it out, and just say something like "... a way to navigate to the next or previous page"

File: docs/topics/pagination.txt [hunk 5]
```diff
@@ -2,15 +2,68 @@
...
+
+            <span class="current">
+                Page {{ page_obj.number }} of {{ paginator.num_pages }}.
+            </span>
+
+            {% if page_obj.has_next %}
+                <a href="?page={{ page_obj.next_page_number }}">next</a>
+            {% endif %}
+        </span>
+    </div>
```
Comment (timmartin) [inline docs/topics/pagination.txt]: This is pretty much an exact duplicate of the example `list.html` at the bottom of the page. This seems like a waste: either there could be one example and both sections link to it, or there could be an initial simple example and a second more complex example.

**Comment by carltongibson** [comment]
Hi @luto. Are you up for finishing this PR? (It looks nearly ready to go: just a rebase and a few comments to address...) Let me know.

Thanks for the input.

**Pull request closed without merging**